### ISSUE TITLE
[A11y] make reason why add-to-basket button is disabled available to keyboard users

### ISSUE THREAD

**Issue by wiffbi** [pretix/pretix#2210]
**Describe the bug**
Currently the add-to-basket button is disabled if no product has a quantity set. The explanation why that button is disabled is only shown on hover of that button. Due to being disabled, keyboard users never reach that button and thus never get any feedback as to why they cannot use that button.

**Expected behavior**
Make explanation available to keyboard users.

**Comment by wiffbi** [comment]
The more I research the subject, it is clear that disabling a button is not ideal regarding accessibility. There is no way for the user to focus the button or otherwise submit the form to get any hint as of why he can’t submit the form. In our case it is rather obvious, but making it explicitly clear through an error-message will be better. Furthermore the current design of the disabled button does not meet the contrast criteria. To meet contrast criteria, the button would either look to similar to an active button or it has to look completely different, which is irritating as well.

TLDR; I would try to have the button work normally. Having the form block onsubmit creates the problem, where to show the message, that one needs to add at least one item to the basket. So blocking the click-event on the button explcitly tells us, that the focus is on the button and we can savely show the message (and maybe have read aloud via aria-live?). This still let’s the form be submitted by hitting enter when inside an form-input, but then the normal error-message mechanism kicks in. I think this an okayish compromise.
